scripts/setup.sh

```diff
@@ -146,10 +146,25 @@ fi
 # Install Ansible Galaxy roles
 if command -v ansible-galaxy &> /dev/null ; then
     echo "Updating Ansible Galaxy roles..."
-    as_user ansible-galaxy collection install --force -r "${ROOT_DIR}/roles/requirements.yml" >/dev/null
-    as_user ansible-galaxy role install --force -r "${ROOT_DIR}/roles/requirements.yml" >/dev/null
-    as_user ansible-galaxy collection install --force -i -r "${ROOT_DIR}/config/requirements.yml" >/dev/null
-    as_user ansible-galaxy role install --force -i -r "${ROOT_DIR}/config/requirements.yml" >/dev/null
+    initial_dir="$(pwd)"
+    roles_path="${ROOT_DIR}/roles/galaxy"
+    collections_path="${ROOT_DIR}/collections"
```

Comment:
Do we need to add this path to our default ansible config? I get a warning while running the setup script:

```
[WARNING]: The specified collections path '/home/dholt/work/deepops/collections' is not part of the configured Ansible collections paths '/home/dholt/.ansible/collections:/usr/share/ansible/collections'. The installed collection
won't be picked up in an Ansible run.
```

Reply:
It's actually already in the config: https://github.com/NVIDIA/deepops/blob/master/ansible.cfg#L2

This is a bit of an ugly issue, and I'm interested in your feedback here.

1. Molecule expects any requirements file to exist within the role directory, so that it can install dependencies as-needed to test.

    So I added `requirements.yml` in the role dir, and specified the requirements file in `molecule/default/molecule.yml`:

    ```
    dependency:
        name: galaxy
        options:
        requirements-file: requirements.yml
    ```
    
    So far, so good.

2. I don't want to specify the same requirements more than once, and it turns out the `requirements.yml` file allows includes! So I rework this a bit so that I can include the dependencies in both our global file and the per-role file:
    
    **roles/singularity_wrapper/roles.yml:**
    ```
    - src: abims_sbr.singularity
      version: 3.7.1-1
    - src: gantsign.golang
      version: 2.4.0
    ```
    
    **roles/singularity_wrapper/requirements.yml:**
    ```
    roles:
    - include: roles.yml
    ```
    
    **roles/requirements.yml:**
    ```
    roles:
    - include: singularity_wrapper/roles.yml
    ```

3. However, when `ansible-galaxy` does includes, [it does them using relative paths at the level of its cwd, not the requirements file](https://github.com/ansible/ansible/issues/46385). Hence the changes to `setup.sh` to cd to the directory the requirements file is in.

4. But lastly... when we do that, the `ansible.cfg` isn't respected any more, because we changed the working directory. So we have to specify our role and collections paths manually, and we get that warning.

Honestly, it's a huge pain, and *almost* led to me abandoning the whole thing. But I like being able to get [individual test results for each role](https://github.com/NVIDIA/deepops/actions/runs/1668722135) 😉  and I think this will add a lot of value in testing when we touch multiple roles at once.

So... I dunno. Can you think of a better way to handle this?